crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -104,25 +104,45 @@ impl AuthorityStorePruner {
         checkpoint_number: CheckpointSequenceNumber,
         metrics: Arc<AuthorityStorePruningMetrics>,
         indirect_objects_threshold: usize,
+        enable_pruning_tombstones: bool,
     ) -> anyhow::Result<()> {
         let _scope = monitored_scope("ObjectsLivePruner");
         let mut wb = perpetual_db.objects.batch();
 
-        let mut object_keys_to_prune = vec![];
+        // Collect objects keys that need to be deleted from `transaction_effects`.
+        let mut live_object_keys_to_prune = vec![];
+        let mut object_tombstones_to_prune = vec![];
         for effects in &transaction_effects {
             for (object_id, seq_number) in effects.modified_at_versions() {
-                object_keys_to_prune.push(ObjectKey(object_id, seq_number));
+                live_object_keys_to_prune.push(ObjectKey(object_id, seq_number));
+            }
+
+            // Indirect object is not being used when pruning tombstone is developed. Therefore, to not further
+            // complicate the code, tombstone pruning is only enabled when indirect object is not turned on.
+            // TODO: re-evaluate the need for indirect object.
+            if enable_pruning_tombstones
+                && (indirect_objects_threshold == 0 || indirect_objects_threshold == usize::MAX)
```

Comment:
can you explain why indirect objects have any effect on tombstone pruning? tombstones themselves cannot be indirect (since they have no contents) so I don't see why there would be any issue here

Reply:
so concern here is not the tombstone object itself, but the proper handling of decrements for other deleted but still visible versions of the same object.
But it should still be fine because those objects were supposed to have been handled by the previous iteration of the pruner, and the proper decrements should already exist in the database. And performing a batch delete in this case for the `objects` table should be ok

Reply:
After discussing with @phoenix-o , it appears that indirect objects are not used. And therefore I just want to be safe and don't want to deal with ref count at all. But I agree that it doesn't really matter here. So I can remove it if it is desirable to do so.

Reply:
yes, let's remove. Ref decrements should be handled correctly even if you do point deletes for predecessors

Reply:
Sounds good. Done. PTAL